pkg/certificates/certs.go

```diff
@@ -170,31 +155,31 @@ func CreateControlPlaneCert(ctx context.Context, caKey *rsa.PrivateKey, caCertif
 }
 
 // CreateDataPlaneCert generates the certificate for the server
-func CreateDataPlaneCert(ctx context.Context, caKey *rsa.PrivateKey, caCertificate *x509.Certificate, expirationInterval time.Duration) (*KeyPair, error) {
+func CreateDataPlaneCert(ctx context.Context, caKey *rsa.PrivateKey, caCertificate *x509.Certificate, expirationInterval time.Duration, san string) (*KeyPair, error) {
 	logger := logging.FromContext(ctx)
 
 	// Then create the private key for the serving cert
-	serverKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	keyPair, err := rsa.GenerateKey(rand.Reader, 2048)
 	if err != nil {
 		logger.Errorw("error generating random key", zap.Error(err))
 		return nil, err
 	}
-	serverCertTemplate, err := createServerCertTemplate(expirationInterval)
+	certTemplate, err := createClientServerCertTemplate(expirationInterval, san)
```

Comment:
Naming is really a bit off overall. Why is this ClientServer? The certs are just used as TLS server certs, right? If we at some point add a client cert for mTLS, we should start talking about client-certs.

Reply:
👍  that a single cert for both client and server seems safe (see above, and usage elsewhere with, for example, service meshes).

I'd recommend fixing the _public_ naming now along with the breaking API change, and leaving the internal function naming to sort out later.